File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int ans = 0;
        int n = s.size();
        int m = t.size();
        int i = 0, j = 0;
        while(j < m)
        {
            if(i >= n)
            {
                ans += m - j;
                return ans;
            }
            while(i < n && s[i] != t[j])
            {
                i++;
            }
            if(i < n && s[i] == t[j])
            {
                i++;
                j++;
            }
        }
        return ans;
    }
};

File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var appendCharacters = function(s, t) {
    let n = s.length;
    let m = t.length;
    let j = 0;
    for(let i = 0; i < n; i++)
    {
        if(s[i] == t[j])
        {
            j++;
        }
    }
    return m - j;
};